crates/services/p2p/src/request_response/messages.rs

```diff
@@ -28,27 +34,38 @@ pub enum RequestMessage {
     Transactions(Range<u32>),
 }
 
-/// Holds oneshot channels for specific responses
-#[derive(Debug)]
-pub enum ResponseChannelItem {
-    Block(oneshot::Sender<Option<SealedBlock>>),
-    SealedHeaders(oneshot::Sender<(PeerId, Option<Vec<SealedBlockHeader>>)>),
-    Transactions(oneshot::Sender<Option<Vec<Transactions>>>),
-}
-
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub enum ResponseMessage {
-    Block(Option<SealedBlock>),
+    Block(Option<Arc<SealedBlock>>),
```

Comment:
Do we really need these `Arc`s? There are a few here and I'm not sure what they're for. Possible it's something I haven't seen yet.

Reply:
Not anymore, at least. Removed in c97533d3